deploy/index.js

```diff
@@ -0,0 +1,374 @@
+      jvmRun: getAssetLink(d.assets, jvmRunReg),
+    };
+  });
+  return names;
+};
+
+/**
+ * Filter all release versions that do not include all asset files
+ *
+ *
+ * @param release - Github release data
+ *
+ * @returns True/False release includes all assets
+ */
+const FilterOnlyCompleteReleases = (release) => {
+  return release && release.jvmRun && release.jar && release.configuration;
+};
+
+/**
+ * Get the tag name of the latest release
+ *
+ * @returns tag name or undefined
+ */
+const getLatestReleaseTagName = async () => {
+  try {
+    let latest = await makeGithubApiLatestRequest();
+    if (latest && latest.status && latest.status === 200 && latest.data) {
+      if (latest.data.tag_name) {
+        return latest.data.tag_name;
+      }
+    }
+    console.err(
+      `Response for latest request unexpected ${JSON.stringify(
+        latest,
+        null,
+        2
+      )}. `
+    );
+  } catch (e) {
+    console.log(e.message);
+    return;
+  }
+};
+
+const transformDownloadUrlToS3Format = (asset) => {
+  const downloadUrlBase = S3_BASE_URL + asset.name + "/";
+  // one way to deep copy
+  let ret = JSON.parse(JSON.stringify(asset));
+  ret.jar.download_url = downloadUrlBase + asset.jar.name;
+  ret.configuration.download_url = downloadUrlBase + asset.configuration.name;
+  ret.jvmRun.download_url = downloadUrlBase + asset.jvmRun.name;
+  return ret;
+};
+
+/**
+ * Request release data and write manifest file.
+ *
+ * returns true for success or false for failure
+ */
+const writeReleaseManifestAndLatest = async () => {
+  let response;
+  try {
+    response = await makeGithubApi();
+  } catch (e) {
+    console.error("Unable to request Github API data.");
+    console.error(e.message);
+    process.exit(-1);
+  }
+  console.log(JSON.stringify(response, null, 3));
+
+  const releases = filterNonFinalReleases(getDataFromResp(response));
+  const res = getReleaseData(sortReleasesByDatePublishDate(releases));
+  const filtered = res.filter(FilterOnlyCompleteReleases);
+  const transformedToS3 = filtered.map(transformDownloadUrlToS3Format);
+
+  writeFileSync(MANIFEST_FILE_NAME, JSON.stringify(transformedToS3, null, 2));
+
+  const latestTagName = await getLatestReleaseTagName();
+
+  let identifyLatestSuccess = false;
+
+  if (latestTagName) {
+    const latestIdx = filtered.findIndex(
+      (release) => release.name === latestTagName
+    );
+    if (latestIdx !== -1) {
+      writeFileSync(
+        LATEST_FILE_NAME,
+        JSON.stringify(transformedToS3[latestIdx], null, 2)
+      );
+      identifyLatestSuccess = true;
+    } else {
+      console.err(
+        `Could not find latest release version tag in releases ${latestTagName}.`
+      );
+    }
+  }
+  //  write the first version in the array !IMPORTANT - releases are sorted in order of release
+  if (!identifyLatestSuccess) {
+    if (filtered.len > 0) {
+      writeFileSync(
+        LATEST_FILE_NAME,
+        JSON.stringify(transformedToS3[0], null, 2)
```

Comment:
magic number/null. What are null and 2 for?

Reply:
changed to const for 2. null is a replacer, it will modify the input object